### ISSUE TITLE
Ability to hide unused accessories from HomeKit

### ISSUE THREAD

**Issue by cozzanez** [yenoiwesa/homebridge-daikin-airbase#30]
Hi, thanks a heap for developing this plugin, it works perfectly. I don’t use fan mode and my AC doesn’t support dry mode. Would it be feasible to introduce the ability hide the accessories associated with these features from being seen by HomeKit?

**Comment by yenoiwesa** [comment]
Hi @cozzanez 👋 

The plugin uses the Airbase's device characteristics information to know what features are available. I receive a flag from the Airbase that states whether the unit supports dry mode and fan mode. So I am surprised that your Airbase returns that your unit supports dry mode when in reality it doesn't 🤔 That could be a misconfiguration of your wifi module potentially? (Although as fas as I know there is zero end user accessible setting).

I can investigate the ability to not create these accessories in the plugin's configuration.

**Comment by cozzanez** [comment]
Hi @yenoiwesa

I just checked and dry mode does work. TBH, because the wall controller displayed no fan icon for this mode I assumed it didn’t work. It would still be nice to hide it. 

Cheers 🍻

**Comment by yenoiwesa** [comment]
Hi @cozzanez 👋 

I added a new `overrides` JSON object to the plugin's configuration that allows you to override Daikin Airbase properties.

Here is an extract of the new README file:

| Parameter                    | Type                    | Default | Note                                                                                                                                                                                                                                                                                                                                                                 |
| ---------------------------- | ----------------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `overrides`                  | Object or null          | `null`  | **Optional** - Allows to override Daikin Airbase property values. If left undefined, property values retrieved from the Airbase will be used. Use at your own risk. See below for supported overrides.                                                                                                                                                               |
| `overrides.fanRateSupported` | Boolean                 | N/A     | **Optional** - Define to override the ability to change the fan speed.                                                                                                                                                                                                                                                                                               |
| `overrides.fanModeSupported` | Boolean                 | `true`  | **Optional** - Define to override the ability to control the aircon's fan mode.                                                                                                                                                                                                                                                                                      |
| `overrides.dryModeSupported` | Boolean                 | N/A     | **Optional** - Define to override the ability to control the aircon's dry mode.                                                                                                                                                                                                                                                                                      |

You can install:
```
npm install -g github:yenoiwesa/homebridge-daikin-airbase#70320904298ccea201b9947d61d3c08dba34a363
```
to use the development branch that includes the change.

Then in your configuration you will have to add this new object:
```json
"overrides": {
    "dryModeSupported": false,
    "fanModeSupported": false
}
```

Note that by default the accessory cache will be used, and the configuration will not be read if you have already assigned the Daikin accessories to your home once. You will have to purge your accessory cache for this setting to take effect (this might undo some of your automations).

Could you please let me know how it goes for you with these new options?
Cheers.

**Comment by cozzanez** [comment]
Hi Yenoiwesa,

Works perfectly! After clearing homebridge accessory cache only the heatercooler and fan speed accessories are available in HomeKit.

Thank you very much for doing this! 🍻
________________________________

**Comment by yenoiwesa** [comment]
Hey @cozzanez that's great to hear!

I published [Release v3.1.0](https://github.com/yenoiwesa/homebridge-daikin-airbase/releases/tag/3.1.0) of the platform on npm.

You can install the released version with:
```
npm install -g homebridge-daikin-airbase@latest
```

Cheers 🍻
